tests/test_random.py

```diff
@@ -1,4 +1,17 @@
 from src.random_number import generate
+from pytest_mock import MockerFixture
+from src import random_number
 
-def test_random() -> None:
-    assert generate() <= 10 and generate() >= 1
+def test_random(mocker: MockerFixture) -> None:
+    # arrange
+    mock_random_return = 4
+    mocker.patch.object(random_number, "generate", return_value=mock_random_return)
+    spy=mocker.spy(random_number,"generate")
+    
+    # act
+    res = random_number.generate()
+
+    # assert
+    assert res <= 10 and res >= 1
+    assert spy.call_count == 1
+    assert spy.spy_return == mock_random_return
```

Comment:
you could use `spy.assert_called_once_with(1, 10)` to check that the call count is 1 and it has been called with the parameters `(1, 10)`

Reply:
Con `spy.assert_called_once_with(1, 10)` mi da degli errori strani. 
invece se utilizzo `spy.called_with(1, 10)` il test va a buon fine
cosa cambia?

Reply:
called once include anche il called_count == 1 e quel "assert_called..." davanti ti risparmia di scrivere "assert "

Reply:
(vanno bene entrambi)